src/main/java/practice/StreamPractice.java

```diff
@@ -59,7 +92,14 @@ public Double getOddNumsAverage(List<Integer> numbers) {
      * return the names of all cats whose owners are women from `femaleAge` years old inclusively.
      */
     public List<String> getCatsNames(List<Person> peopleList, int femaleAge) {
-        return Collections.emptyList();
+        return peopleList
+                .stream()
+                .filter(i -> i.getSex() == Person.Sex.WOMAN && i.getAge() >= femaleAge)
+                .map(Person::getCats)
+                .flatMap(Collection::stream)
```

Comment:
```suggestion
                .flatMap(p -> p.getCatList().stream())
```

lets combine map and flatMap here